android/src/main/kotlin/br/com/adriankohls/zendesk2/Zendesk2Chat.kt

```diff
@@ -179,7 +187,7 @@ class Zendesk2Chat(private val activity: Activity?, private val channel: MethodC
         chatProviderObservationToken = ObservationScope()
         Chat.INSTANCE.providers()?.chatProvider()?.observeChatState(chatProviderObservationToken!!) {
             this.agents = it.agents
-            this.hasAgents = it.agents.isNotEmpty()
```

Comment:
You should revert this too.

these observer updated connection status and if it has online agents, doesn't mean that chat is connected that it will always have online agents, this is wrong